test/boost/commitlog_test.cc

```diff
@@ -305,23 +305,20 @@ SEASTAR_TEST_CASE(test_commitlog_delete_when_over_disk_limit) {
     return cl_test(cfg, [](commitlog& log) {
             auto sem = make_lw_shared<semaphore>(0);
             auto segments = make_lw_shared<segment_names>();
+            auto nn = make_lw_shared<size_t>(0);
 
             // add a flush handler that simply says we're done with the range.
-            auto r = log.add_flush_handler([&log, sem, segments](cf_id_type id, replay_position pos) {
-                auto f = make_ready_future<>();
-                // #6195 only get segment list at first callback. We can (not often)
-                // be called again, but reading segment list at that point might (will)
-                // render same list as in the diff check below. 
-                if (segments->empty()) {
-                    *segments = log.get_active_segment_names();
-                    // Verify #5899 - file size should not exceed the config max. 
-                    f = parallel_for_each(*segments, [](sstring filename) {
-                        return file_size(filename).then([](uint64_t size) {
-                            BOOST_REQUIRE_LE(size, max_size_mb * 1024 * 1024);
-                        });
+            auto r = log.add_flush_handler([&log, sem, segments, nn](cf_id_type id, replay_position pos) -> future<> {
+                auto diff = segment_diff(log, *segments);
```

Comment:
This is a bit unintuitive, as in the best case, doing a diff here is pointless, since segments are not removed until _after_ this callback. Of course, you do a final diff later on, but still. Needs a comment. (Stone, glasshouse, here I am)
Would it not be better to make `segments` a  set and simply add to it in every callback, thus giving a view of all segments that have existed during the CL lifetime, and then compare it against those surviving and the deletion count below?

Reply:
I think the solution with a set sounds best, I'm adding it in the rebase